packages/api/tests/models/test-rules-model.js

```diff
@@ -149,10 +161,27 @@ test('Creating a rule with an invalid type throws an error', async (t) => {
 
   await t.throwsAsync(
     () => rulesModel.create(ruleItem),
-    { message: 'Rule type \'invalid\' not supported.' }
+    { name: 'SchemaValidationError' }
   );
 });
 
+test.serial('Creating an invalid rule does not create workflow triggers', async (t) => {
+  const { onetimeRule } = t.context;
+  const ruleItem = cloneDeep(onetimeRule);
+
+  ruleItem.rule.type = 'invalid';
+
+  const createTriggerStub = sinon.stub(models.Rule.prototype, 'createRuleTrigger').resolves(ruleItem);
+
+  await t.throwsAsync(
+    () => rulesModel.create(ruleItem),
+    { name: 'SchemaValidationError' }
+  );
+
+  t.true(createTriggerStub.notCalled);
+  createTriggerStub.restore();
```

Comment:
```suggestion
  const createTriggerStub = sinon.stub(models.Rule.prototype, 'createRuleTrigger').resolves(ruleItem);

  try {
    await t.throwsAsync(
      () => rulesModel.create(ruleItem),
      { name: 'SchemaValidationError' }
    );
    t.true(createTriggerStub.notCalled);
  } finally {
    createTriggerStub.restore();
  }
```

We want to make sure that this stub gets restored even if the test fails on `throwsAsync`, so that the stub isn't still active when other tests run. We also want to do this for the `update` test